nixos/modules/programs/environment.nix

```diff
@@ -21,6 +21,8 @@ in
         NIXPKGS_CONFIG = "/etc/nix/nixpkgs-config.nix";
         PAGER = mkDefault "less -R";
         EDITOR = mkDefault "nano";
+        XDG_CONFIG_HOME = "$HOME/.config";
+        XDG_DATA_HOME = "$HOME/.local/share";
```

Comment:
AFAIK, there's some broken software that won't work correctly without explicitly setting them. I don't remember exactly which applications, but I set them explicitly in my config.